### PR TITLE
Updates to latest go and deps, restores "indirect"

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This updates versions to latest with no visible impact, while also
cleaning up the workflow files wrt centralizing the go version.

Before, we accidentally didn't have indirect dependencies in our go.mod
file. This hid why our binary is so large (>12M). This restores the
inconvenient information so that it becomes easier to remove the bulk.

### CODE REVIEW

File: .bingo/Variables.mk [hunk 1]
```diff
@@ -29,11 +29,11 @@ $(GOIMPORTS): $(BINGO_DIR)/goimports.mod
 	@echo "(re)installing $(GOBIN)/goimports-v0.1.0"
 	@cd $(BINGO_DIR) && $(GO) build -mod=mod -modfile=goimports.mod -o=$(GOBIN)/goimports-v0.1.0 "golang.org/x/tools/cmd/goimports"
 
-GOLANGCI_LINT := $(GOBIN)/golangci-lint-v1.39.0
+GOLANGCI_LINT := $(GOBIN)/golangci-lint-v1.40.0
```
Comment (codefromthecrypt) [inline .bingo/Variables.mk:32]: nothing notable here

File: .github/workflows/commit.yaml [hunk 2]
```diff
@@ -12,7 +15,7 @@ on:
     branches:
       - '**'
   # workflow_dispatch will let us manually trigger the workflow from GitHub actions dashboard.
-  # This is important because sometimes the macos build fails due to Docker installation.
+  # For example, you can try to build a branch without raising a pull request.
```
Comment (codefromthecrypt) [inline .github/workflows/commit.yaml:18]: we no longer install docker, but there's another reason to keep this

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -7,11 +7,11 @@ name: "release"
 on:
   push:
     tags:
-      - '**'
-  # workflow_dispatch will let us manually trigger the workflow from GitHub actions dashboard.
-  # This is important because sometimes the macos build fails due to Docker installation.
-  # See https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs/manually-running-a-workflow
-  workflow_dispatch:
+      - '*'
```
Comment (codefromthecrypt) [inline .github/workflows/release.yaml:10]: simpler way to express all tags. we don't need workflow-dispatch on releases as we can force push the tag again in worst case.

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -7,11 +7,11 @@ name: "release"
...
+
+env:
+  GO_VERSION: "1.16.4"
+  DOCKER_REGISTRY: "ghcr.io"
```
Comment (codefromthecrypt) [inline .github/workflows/release.yaml:14]:
```suggestion
```

File: .github/workflows/release.yaml [hunk 5]
```diff
@@ -27,10 +27,10 @@ jobs:
       - name: "Set up Go"
         uses: actions/setup-go@v2
         with:
-          go-version: '1.16.3'
+          go-version: ${{ env.GO_VERSION }}
 
-      - name: "Login into DockerHub"
-        uses: azure/docker-login@v1
+      - name: "Login into Docker Hub"
+        uses: docker/login-action@v1
```
Comment (codefromthecrypt) [inline .github/workflows/release.yaml:33]: less strange

File: .github/workflows/release.yaml [hunk 6]
```diff
@@ -78,7 +78,7 @@ jobs:
           INPUT_FILE: getenvoy_${{ env.RELEASE_VERSION }}_${{ env.OS_NAME }}_x86_64.tar.gz
           INPUT_VERSION: tags/${{ env.RELEASE_TAG }}
         run: |  # extract getenvoy to location used in `make e2e`. don't chmod because tar.gz should be correct.
-          curl -s https://raw.githubusercontent.com/dsaltares/fetch-gh-release-asset/0.0.5/fetch_github_asset.sh | bash
+          curl -s https://raw.githubusercontent.com/dsaltares/fetch-gh-release-asset/0.0.6/fetch_github_asset.sh | bash
```
Comment (codefromthecrypt) [inline .github/workflows/release.yaml:81]: nothing very interesting

File: go.mod [hunk 7]
```diff
@@ -6,16 +6,31 @@ go 1.16
...
-	github.com/tetratelabs/log v0.0.0-20210323000454-90a3a3e141b5
+	github.com/tetratelabs/log v0.0.0-20210422163326-7ba70517903b
 	github.com/tetratelabs/multierror v1.1.0
+	github.com/tklauser/go-sysconf v0.3.5 // indirect
+	github.com/ulikunitz/xz v0.5.10 // indirect
+	go.uber.org/multierr v1.7.0 // indirect
+	golang.org/x/crypto v0.0.0-20210506145944-38f3c27a63bf // indirect
+	golang.org/x/sys v0.0.0-20210510120138-977fb7262007 // indirect
+	golang.org/x/term v0.0.0-20210503060354-a79de5458b56 // indirect
+	google.golang.org/grpc v1.37.0 // indirect
```
Comment (codefromthecrypt) [inline go.mod:34]: horrifying we have a grpc dep still, even if indirect

**Review by codefromthecrypt — COMMENTED** [review]
notes

**Review by mathetake — APPROVED** [review]

**Pull request merged**